vignettes/function-flow.Rmd

```diff
@@ -212,16 +417,106 @@ extracted diabetes population as having either T1D or T2D. As described
 in the `vignette("design")`, individuals not classified as T1D cases are
 classified as T2D cases.
 
-The output is a `data.frame` that includes one row per individual in the
-diabetes population: one column with their PNR, two columns with
-inclusion dates (one "stable" date and one "raw" date - see the
-`vignette("design")` for an elaboration on what that entails), and one
-column with the diabetes type.
-
-<!-- TODO: add a link to the specific section where this is described -->
-
-![Flow of functions for classifying diabetes status using the `osdc`
-package.](images/function-flow-classification.png)
+As the diabetes type classification incorporates an evaluation of the
+time from diagnosis/inclusion to first subsequent purchase of insulin,
+the `get_diabetes_type()` function has to take the date of diagnosis and
+all purchases of GLD drugs (after censoring) as inputs. In addition,
+information on diabetes type-specific primary diagnoses from hospitals
+is also a requirement.
+
+Thus, the function takes the following inputs from
+`get_diagnosis_date()`, `exclude_pregnancy()`, and
+`include_diabetes_diagnoses()`:
+
+-   From `get_diagnosis_date()`: Information on date of diagnosis of
+    diabetes
+    -   `pnr`
+    -   `raw_inclusion_date`
+    -   `stable_inclusion_date`
+-   From `exclude_pregnancy()`: Information on historic GLD purchases:
+    -   `pnr`: identifier variable
+    -   `date`: dates of all purchases of GLD.
+    -   `atc`: type of drug
+    -   `contained_doses`: defined daily doses of drug contained in
+        purchase
+-   From `include_diabetes_diagnoses()`: Information on diabetes
+    type-specific primary diagnoses from hospitals:
+    -   `pnr`: identifier variable
+    -   `n_t1d_endocrinology`: number of type 1 diabetes-specific
+        primary diagnosis codes from endocrinological departments
+    -   `n_t2d_endocrinology`: number of type 2 diabetes-specific
+        primary diagnosis codes from endocrinological departments
+    -   `n_t1d_medical`: number of type 1 diabetes-specific primary
+        diagnosis codes from medical departments
+    -   `n_t2d_medical`: number of type 2 diabetes-specific primary
+        diagnosis codes from medical departments
```

Comment:
Shouldn't these columns be kept in `get_diagnosis_date()` then?

Reply:
That's a question! I've been unsure of how to best handle these outputs from `include_diabetes_diagnoses()` and `exclude_pregnancy()`. Whether to keep these variables in `get_diagnosis_date()` and process them internally in that function (that is probably what I should have done), or pass them "outside" `get_diagnosis_date()` to `get_diabetes_type()` (which is what I've done here).

Now, I much prefer to pass them to `get_diagnosis_date()` and put all the helper functions (`get_only_insulin_purchases()`, `get_insulin_purchases_within_180_days()`, `get_insulin_is_two_thirds_of_gld_doses()`, `get_any_t1d_primary_diagnoses()`, `get_type_diagnoses_from_endocrinology()`, and `get_type_diagnosis_majority()`) inside of `get_diagnosis_date()` and have them generate columns with a boolean variable for each of their outputs. That would be much better, and the output is in a format that is easy to process in `get_diabetes_type()`.

Reply:
I'll revise this part in the other PR's where the documentation and function interfaces have been split up.